client/app/components/TimeAgo.jsx

```diff
@@ -0,0 +1,100 @@
+import moment from 'moment';
+import { isNil } from 'lodash';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { Moment } from '@/components/proptypes';
+import { clientConfig } from '@/services/auth';
+
+const interactiveComponents = new Set();
+
+function updateInteractiveComponents() {
+  interactiveComponents.forEach(component => component.forceUpdate());
+  setTimeout(updateInteractiveComponents, 30 * 1000);
+}
+updateInteractiveComponents();
+
+function toMoment(value) {
+  if (isNil(value)) {
+    return null;
+  }
+  value = moment(value);
+  return value.isValid() ? value : null;
+}
+
+export class TimeAgo extends React.Component {
+  static propTypes = {
+    date: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+      PropTypes.instanceOf(Date),
+      Moment,
+    ]),
+    interactive: PropTypes.bool,
+    placeholder: PropTypes.string,
+  };
+
+  static defaultProps = {
+    date: null,
+    interactive: true,
+    placeholder: '',
+  };
+
+  constructor(props) {
+    super(props);
+    this._value = toMoment(this.props.date);
+  }
+
+  componentDidMount() {
+    if (this.props.interactive) {
+      interactiveComponents.add(this);
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    this._value = toMoment(nextProps.date);
```

Comment:
We call `forceUpdate` every 30 seconds. Does this mean we will call `toMoment` every 30 seconds?

Reply:
How about an implementation that uses state for the formatted value, so we don't need to use `forceUpdate`?

Something like -- https://codesandbox.io/s/jnmjq1p2ny.

Reply:
> We call forceUpdate every 30 seconds. Does this mean we will call toMoment every 30 seconds?

No, `shouldComponentUpdate` is called only when props changed.

> How about an implementation that uses state for the formatted value, so we don't need to use forceUpdate?

Interesting idea, I'll give it a try and if it looks fine - will update PR :+1: